dectest/test.go

```diff
@@ -277,15 +278,15 @@ got   : %v
 func check(t *testing.T, z, r *decimal.Big, c *Case, flags decimal.Condition) {
 	helper(t)()
 	if !equal(z, r) {
-		str := fmt.Sprintf(`%s
-wanted: %q (%s:%d)
-got   : %q (%s:%d)
-`,
+		s := []string{
 			c.ShortString(10000),
-			r, flags, -r.Scale(),
-			z, z.Context.Conditions, -z.Scale(),
-		)
-		t.Log(str)
+			fmt.Sprintf("wanted: %q (%s:%d)", r, flags, -r.Scale()),
+			fmt.Sprintf("got   : %q (%s:%d)", z, z.Context.Conditions, -z.Scale()),
+		}
+		if isInexact(z) && allow1ULPError[c.Op] && equalWithin1ULP(z, r) {
+			s = append(s, "passed within 1 ULP")
+		}
+		t.Log(strings.Join(s, "\n"), "\n")
```

Comment:
I refactored this string builder from one fmt.Sprintf() into strings.Join() to preserve test output (but have the ability to insert the optional ULP line in the middle)